src/m365/tenant/commands/auditlog/auditlog-report.ts

```diff
@@ -0,0 +1,272 @@
+  contentId: string;
+  contentUri: string;
+  contentCreated: string;
+  contentExpiration: string;
+}
+
+interface AuditlogReport {
+  CreationTime: string;
+  Id: string;
+  Workload: string;
+  Operation: string;
+  UserId: string;
+}
+
+enum AuditContentTypes {
+  AzureActiveDirectory = "Audit.AzureActiveDirectory",
+  Exchange = "Audit.Exchange",
+  SharePoint = "Audit.SharePoint",
+  General = "Audit.General",
+  DLP = "DLP.All"
+}
+class TenantAuditlogReportCommand extends Command {
+  private serviceUrl: string = 'https://manage.office.com/api/v1.0';
+  private tenantId: string | undefined;
+  private batchedPromises: any = [];
+  private completeAuditReports: AuditlogReport[] = [];
+
+  public get name(): string {
+    return `${commands.TENANT_AUDITLOG_REPORT}`;
+  }
+
+  public get description(): string {
+    return 'Gets audit logs from the Office 365 Management API';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.startTime= typeof args.options.startTime !== 'undefined';
+    telemetryProps.endTime = typeof args.options.endTime !== 'undefined';
+    telemetryProps.contentType = args.options.contentType;
+    return telemetryProps;
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['CreationTime', 'Operation', 'ClientIP', 'UserId', 'Workload'];
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: (err?: any) => void): void {
+    if (this.verbose) {
+      logger.logToStderr(`Start retrieving Audit Log Report`);
+    }
+
+    this.tenantId = Utils.getTenantIdFromAccessToken(auth.service.accessTokens[auth.defaultResource].value);
+    this.getCompleteAuditReports(args, logger)
+      .then((res: AuditlogReport[]): void => {
+        logger.log(res);
+        if (this.verbose) {
+          logger.logToStderr(chalk.green('DONE'));
+        }
+
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  private getCompleteAuditReports(args: CommandArgs, logger: Logger): Promise<AuditlogReport[]> {
+    return this.startContentSubscriptionifNotActive(args, logger)
+      .then((): Promise<AuditContentList[]> => this.getAuditContentList(args, logger))
+      .then((AuditContentLists: AuditContentList[]): void => this.getBatchedPromises(AuditContentLists, 10))
+      .then((): Promise<void> => { return new Promise<void>((resolve: () => void, reject: (error: any) => void): void => { if (this.batchedPromises.length > 0) this.getBatchedAuditlogData(logger, 0, resolve, reject); }) })
+      .then((): AuditlogReport[] => { return this.completeAuditReports.flat(2) });
+  }
+
+  private startContentSubscriptionifNotActive(args: CommandArgs, logger: Logger): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Check whether subscription is active.`);
+    }
+    let SubscriptionListEndpoint: string = 'activity/feed/subscriptions/list';
```

Comment:
My Bad!! 🤦‍♂️. Not sure how did it a miss that. I really really need a sharp eye like yours. 🤞.

Reply:
Just an FYI, adding to that I also have change the variable name from `SubscriptionListEndpoint ` to `subscriptionListEndpoint` since it is local variable following your suggestions.